### PR TITLE
refactor: Set up Cypress as pnpm workspace (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/N8N-5644/set-up-linting-for-cypress

### CODE REVIEW

File: cypress/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "test:e2e:all": "cross-env E2E_TESTS=true start-server-and-test start http://localhost:5678/favicon.ico 'cypress run --headless'",
+    "start": "cd ..; pnpm start"
+  },
+  "devDependencies": {
+    "@ngneat/falso": "^6.1.0",
+    "cross-env": "^7.0.3",
+    "cypress": "^12.8.1",
+    "cypress-real-events": "^1.7.6",
+    "start-server-and-test": "^1.14.0"
+  }
```
Comment (OlegIvaniv) [inline cypress/package.json:18]: I wonder, if we separate the Cypress packages, would they get installed & cached by `e2e-reusable` install job?

File: .github/workflows/e2e-reusable.yml [hunk 2]
```diff
@@ -141,8 +142,8 @@ jobs:
           # We have to provide custom ci-build-id key to make sure that this workflow could be run multiple times
           # in the same parent workflow
           ci-build-id: ${{ needs.prepare.outputs.uuid }}
-          spec: "/__w/n8n/n8n/cypress/${{ inputs.spec }}"
-          config-file: /__w/n8n/n8n/cypress.config.js
+          spec: '/__w/n8n/n8n/cypress/${{ inputs.spec }}'
+          config-file: /__w/n8n/n8n/cypress/cypress.config.js
```
Comment (OlegIvaniv) [inline .github/workflows/e2e-reusable.yml:146]: The e2e specs are now failing, we might have to add /cypress to the caching path similiar to /packages:
```
- name: Cache pnpm modules
        uses: actions/cache@v3
        with:
          path: |
            /github/home/.cache
            /github/home/.pnpm-store
            ./node_modules
            ./packages
          key: ${{ inputs.cache-key }}
          
```

**Review by OlegIvaniv — CHANGES_REQUESTED** [review]
The e2e pipeline is failing with these changes :/ 
https://github.com/n8n-io/n8n/actions/runs/4764774969/jobs/8469859284 

You can trigger e2e pipeline for this specific branch running `gh workflow run e2e-tests.yml --ref n8n-6365 ` if you have Github CLI installed

**Comment by ivov** [comment]
Marking as draft till next tech debt day

**Comment by ivov** [comment]
To revisit at a later time.

**Review by netroy — APPROVED** [review]
rebased and fixed this up.

[tests also pass now](https://github.com/n8n-io/n8n/actions/runs/9448641273).

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.46.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.46.0)

**Pull request merged**